src/services/profile/profile-router.ts

```diff
@@ -72,4 +76,235 @@ profileRouter.get("/leaderboard/", async (req: Request, res: Response) => {
+
+    if (user) {
+        return res.status(Constants.FAILURE).send({ error: "UserAlreadyExists" });
+    }
+
+    const profileMetadata: AttendeeMetadata = new AttendeeMetadata(profile.userId, Constants.DEFAULT_FOOD_WAVE);
+
+    try {
+        await AttendeeProfileModel.create(profile);
+        await AttendeeMetadataModel.create(profileMetadata);
+    } catch (error) {
+        console.error(error);
+        return res.status(Constants.FAILURE).send({ error: "InvalidParams" });
+    }
+
+    return res.status(Constants.SUCCESS).send(profile);
+});
+
+/**
+ * @api {put} /profile/points PUT /profile/points
+ * @apiGroup Profile
+ * @apiDescription Update a user's points. Only STAFF and ADMIN roles can edit this.
+ *
+ * @apiBody {String} points new number of points
+ *
+ * @apiSuccess (200: Success) {Json} profile Updated user point number
+ * @apiSuccessExample Example Success Response:
+ * HTTP/1.1 200 OK
+ * {
+ *    "points": 5
+ * }
+ *
+ * @apiError (500: Internal Error) {String} InternalError An internal server error occurred.
+ * @apiErrorExample Example Error Response (InternalError):
+ *     HTTP/1.1 500 Internal Server Error
+ *     {"error": "InternalError"}
+ */
+
+profileRouter.put("/points", strongJwtVerification, async (req: Request, res: Response) => {
+    const profile: AttendeeProfile | null = req.body as AttendeeProfile;
+    if (!profile) {
+        return res.status(Constants.BAD_REQUEST).send({ error: "InvalidPutData" });
+    }
+
+    const decodedData: JwtPayload = res.locals.payload as JwtPayload;
+
+    if (!hasElevatedPerms(decodedData)) {
+        return res.status(Constants.FORBIDDEN).send({ error: "NotAuthorizedToUseEndpoint" });
+    }
+
+    // // eslint-disable-next-line @typescript-eslint/typedef
+    const update = {
+        $set: {
+            points: profile.points,
+        },
+    };
+
+    // eslint-disable-next-line @typescript-eslint/typedef
+    const filter = { userId: decodedData.id };
+
+    await AttendeeProfileModel.updateOne(filter, update);
+
+    return res.status(Constants.SUCCESS).send({ points: profile.points });
+});
+
+/**
+ * @api {delete} /profile DELETE /profile
+ * @apiGroup Profile
+ * @apiDescription Delete the user's profile based on their authentication.
+ *
+ * @apiSuccess (200: Success) {Json} success Indicates successful deletion of the user's profile.
+ * @apiSuccessExample Example Success Response:
+ * HTTP/1.1 200 OK
+ * {
+ *    "success": true
+ * }
+ *
+ * @apiError (500: Internal Error) {String} InternalError An internal server error occurred.
+ * @apiErrorExample Example Error Response (InternalError):
+ *     HTTP/1.1 500 Internal Server Error
+ *     {"error": "InternalError"}
+ */
+
+profileRouter.delete("/", strongJwtVerification, async (_: Request, res: Response) => {
+    const decodedData: JwtPayload = res.locals.payload as JwtPayload;
+
+    // eslint-disable-next-line @typescript-eslint/typedef
+    const filter = { userId: decodedData.id };
```

Comment:
Don't use eslint disables, just pass the filter in directly or create a filter type